### ISSUE TITLE
[@typescript-eslint/no-unused-vars] Rule doesn't work with object destructuring and spread operator

### ISSUE THREAD

**Issue by kesupile-kesupile** [typescript-eslint/typescript-eslint#2643]
The `@typescript-eslint/no-unused-vars` rule doesn't work when desctructuring an object while also using the spread operator.

**Repro**
Example:

```ts
export const foo = (v: { x: number; y: number }) => {
  const { x, ...rest } = v;
  return rest.y + 4;
};
```
**Expected result**
There is an error caused by the fact that we destructured x but didn't use it.

**Actual Result**
No error at all. I'm sure the rule _is_ working because I get an error if I do this:
```ts
export const blah = (v: { x: number; y: number }) => {
  const { x } = v;
  return 4;
};
```

**Config**
```js
// .eslintrc.js
{
  "rules": {
    "@typescript-eslint/<rule>": "error"
  }
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `13.8.0` |

**Comment by bradzacher** [comment]
https://eslint.org/docs/rules/no-unused-vars

Please read the rule docs. There's an option you'll want to use.
